notebooks/00-release-28.0.0/03-full-timeline-queries.ipynb

```diff
@@ -0,0 +1,584 @@
+   "execution_count": null,
+   "id": "9f314edb-439d-413b-bbc2-7dd965ad24a1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# start streaming ingestion job\n",
+    "kafka_ingestion_spec = {\n",
+    "  \"type\": \"kafka\",\n",
+    "  \"spec\": {\n",
+    "    \"ioConfig\": { \"type\": \"kafka\",  \"consumerProperties\": { \"bootstrap.servers\": \"kafka:9092\" },\n",
+    "      \"topic\": \"clicks\",\n",
+    "      \"inputFormat\": { \"type\": \"kafka\", \"valueFormat\": { \"type\": \"json\" }   },\n",
+    "       \"useEarliestOffset\": True\n",
+    "    },\n",
+    "    \"tuningConfig\": { \"type\": \"kafka\"  },\n",
+    "    \"dataSchema\": {\n",
+    "      \"dataSource\": \"example-clicks-full-timeline\",\n",
+    "      \"timestampSpec\": { \"column\": \"time\", \"format\": \"iso\" },\n",
+    "      \"dimensionsSpec\": {\n",
+    "        \"dimensions\": [ ],\n",
+    "        \"useSchemaDiscovery\": True\n",
+    "      },\n",
+    "      \"granularitySpec\": {\n",
+    "        \"queryGranularity\": \"none\",\n",
+    "        \"rollup\": False,\n",
+    "        \"segmentGranularity\": \"day\"\n",
+    "      }\n",
+    "    }\n",
+    "  }\n",
+    "}\n",
+    "druid.rest.post(\"/druid/indexer/v1/supervisor\", json.dumps(kafka_ingestion_spec), headers=headers)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "234e253d-00ee-4e74-8e31-4e724cb0c324",
+   "metadata": {},
+   "source": [
+    "### Nothing new so far\n",
+    "\n",
+    "It might take it a minute or two to start making the real-time data available. \n",
+    "There are 3 months of data in the table and the query next query shows when it has caught up to almost now. Run it a few times until you see the full 90+ days in the data and the most recent event within a minute. Events are being generated irregularly, but there should be at a new one every minute. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dd200ed0-5d3e-4d0d-ba0c-f4d310e38133",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "  SELECT \n",
+    "      min(__time) \"min_time\", \n",
+    "      max(__time) \"max_time\", \n",
+    "      TIMESTAMPDIFF( DAY, min(__time), max(__time)) \"days_of_data\", \n",
+    "      TIMESTAMPDIFF( SECOND, CURRENT_TIMESTAMP, max(__time)) \"recent_event_seconds_ago\" \n",
+    "  FROM \"example-clicks-full-timeline\"\n",
+    "'''\n",
+    "display_client.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3bc7e74f-fdd5-4b79-9c83-c74b6142aa69",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "display(datagen.get(f\"/status/gen_clickstream_stream\", '').json())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8662a937-71cc-4b65-95cf-ea2c4943f947",
+   "metadata": {},
+   "source": [
+    "## Retention rules\n",
+    "\n",
+    "Retention rules can be defined to only cache a portion of the timeline on historical servers.\n",
```

Comment:
the docs do not say why you need `useDefaultTierForNull`... so I excluded that comment.

I edited it a bit to describe what is going on. the historical at this stage had already loaded the segments, so it will actually remove them as a result of this. I explained this.